journald: fix vacuuming of archived journals
d_name is modified on line 227 so if the entire journal name is needed
again p must be used. Before this change when journal_file_empty was called
on archived journals it would always return with -2.

Signed-off-by: George McCollister <george.mccollister@gmail.com>

Index: systemd-206/src/journal/journal-vacuum.c
===================================================================
--- systemd-206.orig/src/journal/journal-vacuum.c	2014-01-29 20:43:27.492343295 -0200
+++ systemd-206/src/journal/journal-vacuum.c	2014-01-29 20:48:55.076356876 -0200
@@ -265,18 +265,18 @@
                         /* We do not vacuum active files or unknown files! */
                         continue;
 
-                if (journal_file_empty(dirfd(d), de->d_name)) {
+                if (journal_file_empty(dirfd(d), p)) {
 
                         /* Always vacuum empty non-online files. */
 
-                        if (unlinkat(dirfd(d), de->d_name, 0) >= 0)
-                                log_debug("Deleted empty journal %s/%s.", directory, de->d_name);
+                        if (unlinkat(dirfd(d), p, 0) >= 0)
+                                log_debug("Deleted empty journal %s/%s.", directory, p);
                         else if (errno != ENOENT)
-                                log_warning("Failed to delete %s/%s: %m", directory, de->d_name);
+                                log_warning("Failed to delete %s/%s: %m", directory, p);
                         continue;
                 }
 
-                patch_realtime(directory, de->d_name, &st, &realtime);
+                patch_realtime(directory, p, &st, &realtime);
 
                 GREEDY_REALLOC(list, n_allocated, n_list + 1);


Upstream-Status: Backport [207] 
